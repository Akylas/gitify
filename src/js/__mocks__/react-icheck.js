import React from 'react';

export class Checkbox extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="checkbox"
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class Radio extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="radio"
          value={this.props.value}
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class RadioGroup extends React.PureComponent {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  };
}
