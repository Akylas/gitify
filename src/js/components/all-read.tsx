import * as React from 'react';
import Typist from 'react-typist';
import styled from 'styled-components';
import { emojify } from 'react-emojione';

import constants from '../utils/constants';

const Wrapper = styled.div`
  padding: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Emoji = styled.div`
  margin-top: 2rem;
  font-size: 4rem;
  font-weight: 400;
`;

const Title = styled.h2`
  margin: 0.5rem 0 0;
  padding: 10px 5px;
  font-weight: 600;
  line-height: 40px;
`;

const Desc = styled.h4`
  font-weight: 400;
`;

export const AllRead = () => {
  const message =
    constants.ALLREAD_MESSAGES[
      Math.floor(Math.random() * constants.ALLREAD_MESSAGES.length)
    ];

  const emoji =
    constants.ALLREAD_EMOJIS[
      Math.floor(Math.random() * constants.ALLREAD_EMOJIS.length)
    ];

  return (
    <Wrapper>
      <Emoji>{emojify(emoji, { output: 'unicode' })}</Emoji>

      <Title>
        <Typist>{message}</Typist>
      </Title>

      <Desc>No new notifications.</Desc>
    </Wrapper>
  );
};
