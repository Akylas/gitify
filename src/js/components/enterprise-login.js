const ipcRenderer = require('electron').ipcRenderer;

import React from 'react';
import { Link } from 'react-router-dom';
import { reduxForm, Field } from 'redux-form';
import { connect } from 'react-redux';

import { loginUser } from '../actions';
import { authGithub } from '../utils/helpers';

const renderField = ({ input, label, placeholder, meta: { touched, error, warning } }) => (
  <div className="form-group">
    <label htmlFor={input.name}>{label}</label>
    <div>
      <input {...input} className="form-control" placeholder={placeholder} type="text" />
      {touched && ((error && <span>{error}</span>) || (warning && <span>{warning}</span>))}
    </div>
  </div>
);

export class EnterpriseLogin extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.isLoggedIn) {
      ipcRenderer.send('reopen-window');
      // FIXME! <Redirect /> from react-router
      this.context.router.push('/notifications');
    }
  }

  handleSubmit(data, dispatch) {
    authGithub(data, dispatch(loginUser));
  }

  render() {
    return (
      <div className="container-fluid main-container settings">
        <Link
          to="/login"
          className="btn btn-lg btn-block enterprise"
          replace
        >
          <i className="fa fa-angle-left" /> Back
        </Link>

        <form onSubmit={this.props.handleSubmit(this.handleSubmit)}>
          <Field
            name="hostname"
            component={renderField}
            label="Host Name"
            placeholder="github.company.com" />

          <Field
            name="clientId"
            component={renderField}
            label="Client ID"
            placeholder="123456789" />

          <Field
            name="clientSecret"
            component={renderField}
            label="Client Secret"
            placeholder="ABC123DEF456" />

          <button className="btn btn-md" type="submit">
            <i className="fa fa-github" /> Login to GitHub Enterprise
          </button>
        </form>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    isLoggedIn: state.auth.get('token') !== null,
    settings: state.settings
  };
};

export default connect(mapStateToProps, null)(reduxForm({
  form: 'loginEnterprise',
  initialValues: {
    hostname: 'github.example.com',
    clientId: '123123123',
    clientSecret: 'ABC123ABC123',
  },
})(EnterpriseLogin));
