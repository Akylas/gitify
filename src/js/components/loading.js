import React from 'react';
import { connect } from 'react-redux';
import NProgress from 'nprogress';

export class Loading extends React.PureComponent {
  componentDidMount() {
    NProgress.configure({
      showSpinner: false
    });

    if (this.props.isLoading) {
      NProgress.start();
    }
  }

  componentWillUnmount() {
    NProgress.remove();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.isLoading) {
      NProgress.start();
    } else {
      NProgress.done();
    }
  }

  render() {
    return null;
  };
};


export function mapStateToProps(state) {
  return {
    isLoading: state.notifications.get('isFetching')
  };
};

export default connect(mapStateToProps, null)(Loading);
