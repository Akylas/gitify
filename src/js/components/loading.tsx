import * as React from 'react';
import * as NProgress from 'nprogress';
import { AppState } from '../../types/reducers';
import { connect } from 'react-redux';

interface IProps {
  isLoading: boolean;
}

export class Loading extends React.PureComponent<IProps> {
  state = {};

  componentDidMount() {
    NProgress.configure({
      showSpinner: false,
    });
  }

  static getDerivedStateFromProps(props, state) {
    if (props.isLoading) {
      NProgress.start();
    } else {
      NProgress.done();
    }

    return {};
  }

  componentWillUnmount() {
    NProgress.remove();
  }

  render() {
    return null;
  }
}

export function mapStateToProps(state: AppState) {
  return {
    isLoading: state.notifications.isFetching,
  };
}

export default connect(mapStateToProps, null)(Loading);
