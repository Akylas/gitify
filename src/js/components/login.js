const { ipcRenderer } = require('electron');

import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { authGithub, isUserEitherLoggedIn } from '../utils/helpers';

export class LoginPage extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.isEitherLoggedIn) {
      ipcRenderer.send('reopen-window');
    }
  }

  render() {
    if (this.props.isEitherLoggedIn) {
      return (
        <Redirect to="/" />
      );
    }

    return (
      <div className="container-fluid main-container login">
        <img className="img-responsive logo" src="images/gitify-logo-outline-dark.png" />
        <div className="desc">GitHub Notifications<br />in your menu bar.</div>
        <button
          className="btn btn-block btn-login"
          onClick={() => authGithub(undefined, this.props.dispatch)}
        >
          <i className="fa fa-github mr-2" /> Login to GitHub
        </button>

        <Link
          to="/enterpriselogin"
          className="btn btn-block btn-login mt-3">
          <i className="fa fa-github mr-2" /> Login to GitHub Enterprise
        </Link>
      </div>
    );
  }
};

export function mapStateToProps(state) {
  return {
    isEitherLoggedIn: isUserEitherLoggedIn(state.auth),
  };
};

export default connect(mapStateToProps)(LoginPage);
