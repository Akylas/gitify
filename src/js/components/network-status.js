import React from 'react';

export default class NetworkStatus extends React.Component {
  render() {
    return (
      <div className="network-status">
        <img className="img-fluid logo" src="images/gitify-logo-outline-dark.png" />
        <h4>No Internet Connection</h4>
        <div className="alert alert-danger">Couldn't establish an internet connection.</div>
      </div>
    );
  }
};
