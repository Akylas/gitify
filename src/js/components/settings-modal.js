import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { Checkbox, RadioGroup, Radio } from 'react-icheck';

const ipcRenderer = window.require('electron').ipcRenderer;

import { fetchNotifications, updateSetting, logout, toggleSettingsModal } from '../actions';

export class SettingsModal extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.settings.get('participating') !== this.props.settings.get('participating')) {
      this.props.fetchNotifications();
    }
  }

  onEscape({ keyCode }) {
    if (keyCode === 27) {
      this.props.toggleSettingsModal();
    }
  }

  componentDidMount() {
    document.addEventListener('keydown', (event) => this.onEscape(event));
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', (event) => this.onEscape(event));
  }

  logout() {
    this.props.logout();
    this.context.router.replace('/login');
    ipcRenderer.send('update-icon', 'IconPlain');
  }

  render() {
    const { settings, isOpen } = this.props;

    return (
      <Modal
        className="settings-modal bg-inverse"
        isOpen={isOpen}
        contentLabel="Settings Modal">
        <h3 className="modal-title">
          Settings
          <span
            className="octicon octicon-x btn-close pull-right"
            onClick={() => this.props.toggleSettingsModal()} />
        </h3>

        <div className="row setting">
          <Checkbox
            label="Show only participating"
            checkboxClass="icheckbox_polaris setting-checkbox"
            defaultChecked={settings.get('participating')}
            onChange={(evt) => this.props.updateSetting('participating', evt.target.checked)} />
        </div>

        <div className="row setting">
          <Checkbox
            label="Play sound"
            checkboxClass="icheckbox_polaris setting-checkbox"
            defaultChecked={settings.get('playSound')}
            onChange={(evt) => this.props.updateSetting('playSound', evt.target.checked)} />
        </div>

        <div className="row setting">
          <Checkbox
            label="Show notifications"
            checkboxClass="icheckbox_polaris setting-checkbox"
            defaultChecked={settings.get('showNotifications')}
            onChange={(evt) => this.props.updateSetting('showNotifications', evt.target.checked)} />
        </div>

        <div className="row setting">
          <Checkbox
            label="On Click, Mark as Read"
            checkboxClass="icheckbox_polaris setting-checkbox"
            defaultChecked={settings.get('markOnClick')}
            onChange={(evt) => this.props.updateSetting('markOnClick', evt.target.checked)} />
        </div>

        <div className="row setting">
          <Checkbox
            label="Open at startup"
            checkboxClass="icheckbox_polaris setting-checkbox"
            defaultChecked={settings.get('openAtStartup')}
            onChange={(evt) => this.props.updateSetting('openAtStartup', evt.target.checked)} />
        </div>

      <RadioGroup
        name="showAppIcon"
        value={settings.get('showAppIcon')}
        onChange={(evt) => this.props.updateSetting('showAppIcon', evt.target.value)}>
        <Radio
          value="both"
          radioClass="iradio_polaris setting-radio"
          label="Both Icons" />

        <Radio
          value="tray"
          radioClass="iradio_polaris setting-radio"
          label="Tray Icon" />

        <Radio
          value="dock"
          radioClass="iradio_polaris setting-radio"
          label="Dock Icon" />
      </RadioGroup>

      <button className="btn btn-block btn-outline-danger btn-logout" onClick={() => this.logout()}>
        <span className="octicon octicon-sign-out" /> Logout
      </button>
    </Modal>
    );
  }
};

SettingsModal.contextTypes = {
  router: React.PropTypes.object.isRequired
};

SettingsModal.propTypes = {
  isOpen: React.PropTypes.bool.isRequired
};

function mapStateToProps(state) {
  return {
    settings: state.settings
  };
};

export default connect(mapStateToProps, {
  updateSetting,
  fetchNotifications,
  logout,
  toggleSettingsModal
})(SettingsModal);
