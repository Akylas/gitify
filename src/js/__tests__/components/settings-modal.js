import React from 'react'; // eslint-disable-line no-unused-vars
import { Map } from 'immutable';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Checkbox, RadioGroup, Radio } from 'react-icheck';

import { SettingsModal } from '../../components/settings-modal';
const ipcRenderer = window.require('electron').ipcRenderer;

const options = {
  context: {
    location: {
      pathname: ''
    },
    router: {
      push: sinon.spy(),
      replace: sinon.spy()
    }
  }
};

function setup(props) {
  const wrapper = shallow(<SettingsModal {...props} />, options);

  return {
    context: options.context,
    props: props,
    wrapper: wrapper,
  };
};

describe('components/settings-modal.js', function () {

  beforeEach(function() {
    ipcRenderer.send.reset();
  });

  it('should render itself & its children', function () {

    const props = {
      isOpen: true,
      updateSetting: sinon.spy(),
      fetchNotifications: sinon.spy(),
      logout: sinon.spy(),
      settings: Map({
        participating: false,
        playSound: true,
        showNotifications: true,
        markOnClick: false,
        openAtStartup: false
      })
    };

    const { wrapper } = setup(props);
    expect(wrapper).to.exist;
    expect(wrapper.find(Checkbox).length).to.equal(5);
    expect(wrapper.find(RadioGroup).length).to.equal(1);
    expect(wrapper.find(Radio).length).to.equal(3);
    expect(wrapper.find('.octicon-sign-out').length).to.equal(1);

  });

  it('should update a setting', function () {

    const props = {
      isOpen: true,
      updateSetting: sinon.spy(),
      fetchNotifications: sinon.spy(),
      logout: sinon.spy(),
      settings: Map({
        participating: false,
        playSound: true,
        showNotifications: true,
        markOnClick: false,
        openAtStartup: false
      })
    };

    const { wrapper } = setup(props);
    expect(wrapper).to.exist;

    // Note: First Check is "participating"
    wrapper.find(Checkbox).first().props().onChange({target: {checked: true}});
    expect(props.updateSetting).to.have.been.calledOnce;

    wrapper.setProps({
      ...props,
      settings: Map({
        ...props.settings,
        participating: true
      })
    });
    expect(props.fetchNotifications).to.have.been.calledOnce;

  });

  it('should press the logout', function () {

    const props = {
      isOpen: true,
      updateSetting: sinon.spy(),
      fetchNotifications: sinon.spy(),
      logout: sinon.spy(),
      settings: Map({
        participating: false,
        playSound: true,
        showNotifications: true,
        markOnClick: false,
        openAtStartup: false
      })
    };

    const { wrapper, context } = setup(props);

    expect(wrapper).to.exist;

    wrapper.find('.octicon-sign-out').parent().simulate('click');

    expect(props.logout).to.have.been.calledOnce;

    expect(ipcRenderer.send).to.have.been.calledOnce;
    expect(ipcRenderer.send).to.have.been.calledWith('update-icon', 'IconPlain');

    expect(context.router.replace).to.have.been.calledOnce;
    expect(context.router.replace).to.have.been.calledWith('/login');

    context.router.replace.reset();

  });

});
