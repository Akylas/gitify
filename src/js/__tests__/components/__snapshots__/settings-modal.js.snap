// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/settings-modal.js should render itself & its children (closed modal) 1`] = `null`;

exports[`components/settings-modal.js should render itself & its children (open modal) 1`] = `
<div>
  <h3
    className="modal-title"
  >
    Settings
    <span
      className="octicon octicon-x btn-close pull-right"
      onClick={[Function]}
    />
  </h3>
  <div
    className="row setting"
  >
    <div>
      <strong>
        Show only participating
      </strong>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="row setting"
  >
    <div>
      <strong>
        Play sound
      </strong>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="row setting"
  >
    <div>
      <strong>
        Show notifications
      </strong>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="row setting"
  >
    <div>
      <strong>
        On Click, Mark as Read
      </strong>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="row setting"
  >
    <div>
      <strong>
        Open at startup
      </strong>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div>
    <div>
      <strong>
        Both Icons
      </strong>
      <input
        checked={undefined}
        onChange={[Function]}
        type="radio"
        value="both"
      />
    </div>
    <div>
      <strong>
        Tray Icon
      </strong>
      <input
        checked={undefined}
        onChange={[Function]}
        type="radio"
        value="tray"
      />
    </div>
    <div>
      <strong>
        Dock Icon
      </strong>
      <input
        checked={undefined}
        onChange={[Function]}
        type="radio"
        value="dock"
      />
    </div>
  </div>
  <button
    className="btn btn-block btn-outline-danger btn-logout"
    onClick={[Function]}
  >
    <span
      className="octicon octicon-sign-out"
    />
     Logout
  </button>
</div>
`;
